### PR TITLE
add list customer information

### PR DESCRIPTION
**Pull request by liyingxin97**
add list customer information and test case

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -34,6 +34,27 @@ def index():
...
+######################################################################
+@app.route("/customers", methods=["GET"])
+def list_customers():
+    """Returns all of the customers"""
+    app.logger.info("Request for customer list")
+    customers = []
+    category = request.args.get("category")
+    name = request.args.get("name")
+    if category:
+        customers = Customer.find_by_category(category)
```
Comment (ryanjcohen) [inline service/routes.py:48]: I think category might be a property that's exclusive to pets, I don't think this field applies in this case

File: service/routes.py [hunk 2]
```diff
@@ -34,6 +34,27 @@ def index():
...
+    elif name:
+        customers = Customer.find_by_name(name)
```
[the remark this answers is not in the document]
Comment (domejiraphon) [reply service/routes.py:50]: I think in our case, we should return every customer added to the database for this function. No needed to filter out something as we don't group customers by some specific category as in Pet.

File: service/routes.py [hunk 3]
```diff
@@ -34,6 +34,27 @@ def index():
...
+######################################################################
+# LIST ALL CUSTOMER
+######################################################################
+@app.route("/customers", methods=["GET"])
+def list_customers():
+    """Returns all of the customers"""
+    app.logger.info("Request for customer list")
+    customers = []
+    category = request.args.get("category")
+    name = request.args.get("firstname")
```
Comment (ryanjcohen) [inline service/routes.py:46]: I don't think this is the correct name for the first name field

File: service/routes.py [hunk 4]
```diff
@@ -58,7 +69,7 @@ def update_customers(customer_id):
 ######################################################################
 # GET Information About the Service
 ######################################################################
-@app.route("/customers", methods=["GET"])
+@app.route("/services", methods=["GET"])
```
[the remark this answers is not in the document]
Comment (Bobby-Zien) [reply service/routes.py:72]: I think its okay to leave it as "/customers"

File: tests/test_routes.py [hunk 5]
```diff
@@ -170,9 +180,17 @@ def test_delete_customer(self):
...
+      #  resp = self.app.get("/customers")
+      #  self.assertEqual(resp.status_code, status.HTTP_200_OK)
+     #   data = resp.get_json()
+     #   self.assertEqual(data["name"], "API_list")
+
+
     def test_services_list(self):
         """ Test services list call """
-        resp = self.app.get("/customers")
+        resp = self.app.get("/services")
```
Comment (Bobby-Zien) [inline tests/test_routes.py:193]: I don't think you should change it

File: service/routes.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Bobby-Zien) [reply service/routes.py:72]: Right right. Thought we are returning all the customers here

**Pull request closed without merging**